spring-core/src/main/kotlin/org/springframework/core/MethodParameterKotlin.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.core
+
+import java.lang.reflect.Constructor
+import java.lang.reflect.Method
+import kotlin.reflect.KParameter
+import kotlin.reflect.jvm.kotlinFunction
+
+/**
+ * Helper internal class that provides Kotlin-specific extensions to [MethodParameter].
+ *
+ * @author Raman Gupta
+ * @since 4.4
+ * @see MethodParameter
+ */
+object MethodParameterKotlin {
```

Comment:
Could we turn this into a regular java `KotlinUtils` class with maybe an additional method to know if Kotlin is present? I think I would like to restrict Kotlin code to only test classes.